__tests__/sum.test.js

```diff
@@ -0,0 +1,68 @@
+const Q = require('q');
+const request = Q.denodeify(require('request'));
+
+const MockClient = require('../src/client/Client');
+
+describe('examples', () => {
+	it('simple mock', done => {
+		return request('http://localhost:2021/simple-mock-example', {json: true}).spread((response, body) => {
+			expect(response.statusCode).toBe(200);
+			expect(body).toEqual({someKey: 'some value'});
+		})
+		.then(done, done.fail);
+	});
+
+	it('dynamic mock', done => {
```

Comment:
I'd like you to add a scenario when we explicitly **re-add** the same route while let it return different data in the second re-adding. Then we call it to check if the response changes after the second re-adding. I think the scenario here you added is also important to have and test. But it seems to be different (it's like the route to have the state for this scenario you are testing.) from what I described above, which is the real use case we will be using in e2e.